flood_api/dependencies/queryparams.py

```diff
@@ -1,23 +1,56 @@
 from datetime import date
-from typing import Annotated
+from typing import Annotated, Optional, Union
 
-from fastapi import Depends, Query
+from fastapi import Depends, HTTPException, Query
 
+from flood_api.settings import settings
+from flood_api.utils.validation_helpers import (
+    validate_bounding_box,
+    validate_coordinates,
+    validate_dates,
+)
 
-def coordinates(
-    lon: Annotated[
-        float,
-        Query(description="Longitude of the coordinate to retrieve data for"),
-    ],
-    lat: Annotated[
-        float,
-        Query(description="Latitude of the coordinate to retrieve data for"),
-    ],
-) -> (float, float):
-    return lon, lat
+INVALID_STATUS_CODE = settings.invalid_query_status_code
 
+# Define a union type for the dependency
+LocationQuery = Union[tuple[float, float], tuple[float, float, float, float]]
 
-CoordinatesDep = Annotated[tuple[float, float], Depends(coordinates)]
+
+def location_query_dependency(
+    lon: Annotated[float | None, Query(description="Longitude")] = None,
+    lat: Annotated[float | None, Query(description="Latitude")] = None,
+    min_lon: Annotated[float | None, Query(description="Minimum longitude")] = None,
+    max_lon: Annotated[float | None, Query(description="Maximum longitude")] = None,
+    min_lat: Annotated[float | None, Query(description="Minimum latitude")] = None,
+    max_lat: Annotated[float | None, Query(description="Maximum latitude")] = None,
+) -> LocationQuery:
+    coordinates = (lat, lon) if None not in (lat, lon) else None
```

Comment:
Subjective opinion, so resolve this comment if you disagree.

I am not a such a big fan of these one-line if-sentences most of the time. I think this is easier to read

```python
if None not in (lat, lon):
    coordinates = (lat, lon)
else:
    coordinates = None
```